create.ps1

```diff
@@ -1,258 +1,449 @@
+        if (-not [string]::IsNullOrEmpty($ErrorObject.ErrorDetails.Message)) {
+            $httpErrorObj.ErrorDetails = $ErrorObject.ErrorDetails.Message
         }
-        else {
-            Write-Output ($Response.access_token)
+        elseif ($ErrorObject.Exception.GetType().FullName -eq 'System.Net.WebException') {
+            if ($null -ne $ErrorObject.Exception.Response) {
+                $streamReaderResponse = [System.IO.StreamReader]::new($ErrorObject.Exception.Response.GetResponseStream()).ReadToEnd()
+                if (-not [string]::IsNullOrEmpty($streamReaderResponse)) {
+                    $httpErrorObj.ErrorDetails = $streamReaderResponse
+                }
+            }
         }
-    }
-    catch {
-        $PSCmdlet.ThrowTerminatingError($PSItem)
-    }
-}
+        try {
 
+            $errorObjectConverted = $ErrorObject.ErrorDetails.Message | ConvertFrom-Json -ErrorAction Stop
 
-function Resolve-ErrorMessage {
-    [CmdletBinding()]
-    param (
-        [Parameter(Mandatory, ValueFromPipeline)]
-        [object]
-        $ErrorObject
-    )
+            if ($null -ne $errorObjectConverted.Error) {
+                if ($null -ne $errorObjectConverted.Error.Message) {
+                    $httpErrorObj.FriendlyMessage = $errorObjectConverted.Error.Message
 
-    process {
-        $Exception = [PSCustomObject]@{
-            FullyQualifiedErrorId = $ErrorObject.FullyQualifiedErrorId
-            MyCommand             = $ErrorObject.InvocationInfo.MyCommand
-            RequestUri            = $ErrorObject.TargetObject.RequestUri
-            ScriptStackTrace      = $ErrorObject.ScriptStackTrace
-            ErrorMessage          = $Null
-            VerboseErrorMessage   = $Null
+                    if ($null -ne $errorObjectConverted.Error.Code) { 
+                        $httpErrorObj.FriendlyMessage = $httpErrorObj.FriendlyMessage + ". Error code: $($errorObjectConverted.Error.Code)"
+                    }
+
+                    if ($null -ne $errorObjectConverted.ErrorDetails) { 
+                        $httpErrorObj.FriendlyMessage = $httpErrorObj.FriendlyMessage + ". Additional details: $($errorObjectConverted.ErrorDetails | ConvertTo-Json)"
+                    }
+                }
+                else {
+                    $httpErrorObj.FriendlyMessage = $errorObjectConverted.Error
+                }
+            }
+            else {
+                $httpErrorObj.FriendlyMessage = $ErrorObject
+            }
+        }
+        catch {
+            $httpErrorObj.FriendlyMessage = $httpErrorObj.ErrorDetails
         }
+        Write-Output $httpErrorObj
+    }
+}
 
-        switch ($ErrorObject.Exception.GetType().FullName) {
-            "Microsoft.PowerShell.Commands.HttpResponseException" {
-                $Exception.ErrorMessage = $ErrorObject.ErrorDetails.Message
-                break
+function Convert-StringToBoolean($obj) {
```

Comment:
Deze functie voelt een beetje te complex aan voor wat het doet. Zeker ook vanwege de recursiviteit. 
Mogelijk kunnen we hier beter gebruik maken van: `[System.Convert]::ToBoolean($value)`. Het zijn ook maar twee boolean value's in de fieldMapping zo te zien.

Reply:
Eens. Dit is nu een algemene functie, die zo complex dient te zijn omdat deze ook met arrays en subobject moet kunnen werken.
Voor deze connector heb ik dit teruggebracht tot het noodzakelijke, maar het lijkt me goed dat we hier met een functie voor komen die alles ondersteund.
Of verschillende functies voor verschillende scenarios.